lldb/packages/Python/lldbsuite/test/decorators.py

```diff
@@ -1010,12 +1024,21 @@ def is_toolchain_with_mte():
             int main() {
                 void* ptr = __arm_mte_create_random_tag((void*)(0), 0);
             }"""
-        cmd = "echo '%s' | %s -march=armv8.5-a+memtag -x c -o %s -" % (
-            test_src,
-            compiler_path,
-            f.name,
+        with open(f_src.name, "w") as f:
+            f.write(test_src)
```

Comment:
Considered using the `subprocess` module and passing the contents through the `input` argument?